scheduler/src/cook/mesos/mesos_mock.clj

```diff
@@ -0,0 +1,430 @@
+         exit-chan (async/chan)
+         start-chan (async/chan)
+         driver (reify SchedulerDriver
+                  (declineOffer [this offer-id]
+                    (log/debug "Declining offer" {:offer-id offer-id})
+                    (async/>!! action-chan [:decline (:value (mesos-type/pb->data offer-id))])
+                    Protos$Status/DRIVER_RUNNING)
+                  (join [this]
+                    (async/<!! exit-chan)
+                    Protos$Status/DRIVER_RUNNING)
+                  (stop [this]
+                    (async/close! exit-chan))
+                  (killTask [this task-id]
+                    (log/debug "In driver killTask" {:task-id task-id :pb->data (mesos-type/pb->data task-id) :value (:value (mesos-type/pb->data task-id))})
+                    (async/>!! action-chan [:kill-task (:value (mesos-type/pb->data task-id))])
+                    Protos$Status/DRIVER_RUNNING)
+                  (^Protos$Status launchTasks [this ^java.util.Collection offer-ids ^java.util.Collection tasks]
+                    (log/debug "Launch tasks called" {:tasks tasks :offer-ids offer-ids})
+                    (async/>!! action-chan [:launch {:offer-ids (map (comp :value mesos-type/pb->data) offer-ids)
+                                                     :tasks (map mesos-type/pb->data tasks)}])
+                    Protos$Status/DRIVER_RUNNING)
+                  (reconcileTasks [this statuses]
+                    (async/>!! action-chan [:reconcile statuses])
+                    Protos$Status/DRIVER_RUNNING)
+                  (start [this]
+                    (async/close! start-chan)
+                    Protos$Status/DRIVER_RUNNING))
+         hosts (->> hosts
+                    (map #(assoc % :available-resources (:resources %)))
+                    (map #(update % :slave-id str)))]
+     (async/thread
+       (try
+         (async/<!! start-chan)
+         (.registered scheduler
+                      driver
+                      (mesos-type/->pb :FrameworkID {:value "cool-framework-id"})
+                      (mesos-type/->pb :MasterInfo {:id "silly id"
+                                                    :ip 127
+                                                    :port 5050
+                                                    :version "1.0.1"}))
+         (loop [state {:config {:task->runtime-ms task->runtime-ms
+                                :task->complete-status task->complete-status
+                                :speed-multiplier speed-multiplier}
+                       :slave-id->host (map-from-vals :slave-id hosts)
+                       :offer-id->offer {}
+                       :task-id->task {}
+                       :task-id->completed-task {} ;;TODO test this in kill
+                       :complete-chan->task-id {}}
+                new-offer-chan (async/timeout (/ 1000 speed-multiplier))]
```

Comment:
Since we have `(/ 1000 speed-multiplier)` duplicated here and below, can we move it up into the outer `let` and call it something like `new-offer-delay-ms`?